server/scripts/availability_dump.js

```diff
@@ -59,6 +61,80 @@ function removeNullPropertiesStream() {
   });
 }
 
+/**
+ * A stream that emits buffers of a given size (in bytes). Useful when an input
+ * stream might emit data in small chunks, while the next stream in a pipeline
+ * works more efficiently with larger chunks, or chunks of a given size.
+ *
+ * In practice, we use this for gzipping, since gzip stream work most
+ * efficiently if data comes in chunks of at least 32 kB (this actually makes
+ * a pretty big difference).
+ *
+ * We couldn't find a good version of this on NPM (which seems surprising,
+ * we're probably missing it). The `stream-chunker` package, which is popular,
+ * performs *terribly* and is often worse than no chunking at all.
+ */
+class BufferedStream extends stream.Transform {
+  constructor({ size = 256 * 1024, ...options } = {}) {
+    super(options);
+    this.size = size;
+    this.resetBuffer();
+  }
+
+  resetBuffer() {
+    this.buffer = Buffer.allocUnsafe(this.size);
```

Comment:
nit: If I correctly understand the (lack of) guarantee `allocUnsafe` makes–that the buffer might already contain data, so you should be careful with reads–we only ever need to allocate a new one here if `this.buffer` is `null` or `undefined`.

Reply:
Oh, good point! This really could just happen once in the constructor, and would be more efficient. That leaves the `this.offset = 0;` line as the only thing in the `resetBuffer()` method, so we might as well just drop the method.

I’ll do that before merging later today.